pkg/satellite/validation_test.go

```diff
@@ -0,0 +1,504 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package satellite
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestValidationUserInfo(t *testing.T) {
```

Comment:
Read documents on table driven tests, this func is quite ideal for it, however the implementation needs work. https://github.com/golang/go/wiki/TableDrivenTests. Also, write test for validation and user info separately, that way you have much less cases.

Reply:
For example: https://golang.org/src/net/mail/message_test.go#L127